agent/provider/src/rules.rs

```diff
@@ -62,30 +65,84 @@ impl RulesManager {
 
     pub fn remove_dangling_rules(&self) -> Result<()> {
         let mut deleted_partner_rules = vec![];
+        let mut deleted_audited_payload_rules = vec![];
 
         let keystore_certs = self.keystore.list_ids();
 
+        let mut rulestore = self.rulestore.config.write().unwrap();
+
+        rulestore.outbound.partner.retain(|cert_id, _| {
+            keystore_certs
+                .contains(cert_id)
+                .not()
+                .then(|| deleted_partner_rules.push(cert_id.clone()))
+                .is_none()
+        });
+
+        rulestore.outbound.audited_payload.retain(|cert_id, _| {
+            keystore_certs
+                .contains(cert_id)
+                .not()
+                .then(|| deleted_audited_payload_rules.push(cert_id.clone()))
+                .is_none()
+        });
+
+        drop(rulestore);
+
+        if !deleted_partner_rules.is_empty() {
+            log::warn!("Because Keystore didn't have appriopriate certs, following Partner rules were removed: {:?}", deleted_partner_rules);
+            self.rulestore.save()?;
+        }
+        if !deleted_audited_payload_rules.is_empty() {
+            log::warn!("Because Keystore didn't have appriopriate certs, following Audited-Payload rules were removed: {:?}", deleted_audited_payload_rules);
+            self.rulestore.save()?;
+        }
+
+        Ok(())
+    }
+
+    pub fn set_audited_payload_mode(&self, cert_id: String, mode: Mode) -> Result<()> {
+        let cert_id = {
+            let certs: Vec<Cert> = self
+                .keystore
+                .list()
+                .into_iter()
+                .filter(|cert| cert.id().starts_with(&cert_id))
+                .collect();
+
+            if certs.is_empty() {
+                bail!(
+                    "Setting Audited-Payload mode {mode} failed: No cert id: {cert_id} found in keystore"
+                );
+            } else if certs.len() > 1 {
+                bail!(
+                    "Setting Audited-Payload mode {mode} failed: Cert id: {cert_id} isn't unique"
+                );
+            } else {
+                let cert = &certs[0];
+                match cert {
+                    Cert::X509(X509CertData { id, .. }) => id.clone(),
+                    Cert::Golem { id, .. } => id.clone(),
```

Comment:
I believe this should return error so golem ones are not valid for audited-payload right now (as we don't validate their permissions by lib)

Reply:
Fixed.